overlord/devicestate/handlers_install.go

```diff
@@ -536,61 +494,70 @@ func (p *preseedSnapHandler) HandleAndDigestAssertedSnap(name, path string, essT
 	return targetPath, sha3_384, uint64(size), nil
 }
 
-var maybeApplyPreseededData = func(st *state.State, ubuntuSeedDir, sysLabel, writableDir string) (preseeded bool, err error) {
-	preseedArtifact := filepath.Join(ubuntuSeedDir, "systems", sysLabel, "preseed.tgz")
-	if !osutil.FileExists(preseedArtifact) {
+func maybeApplyPreseededData(model *asserts.Model, ubuntuSeedDir, sysLabel, writableDir string) (preseeded bool, err error) {
+	sysSeed, err := seedOpen(ubuntuSeedDir, sysLabel)
+	if err != nil {
+		return false, err
+	}
+	preseedSeed := sysSeed.(seed.PreseedCapable)
```

Comment:
Maybe it would be good to add a comment saying this should always work as this is called only for UC20+